worker/src/email.rs

```diff
@@ -0,0 +1,207 @@
+        sec.0.into()
+    }
+}
+
+impl EmailMessage {
+    pub fn new(from: &str, to: &str, raw: &str) -> Result<Self, JsValue> {
+        Ok(EmailMessage(EmailMessageExt::new(from, to, raw)?))
+    }
+
+    // method from is renamed compared to the JS version, because `from` conflicts with the From trait
+    // to was also renamed for consistency
+    pub fn from_envelope(&self) -> String {
+        self.0.from()
+    }
+
+    pub fn to_envelope(&self) -> String {
+        self.0.to()
+    }
+}
+
+pub struct ForwardableEmailMessage(ForwardableEmailMessageExt);
+
+unsafe impl Send for ForwardableEmailMessage {}
+unsafe impl Sync for ForwardableEmailMessage {}
+
+impl JsCast for ForwardableEmailMessage {
+    fn instanceof(val: &JsValue) -> bool {
+        val.is_instance_of::<ForwardableEmailMessageExt>()
+    }
+
+    fn unchecked_from_js(val: JsValue) -> Self {
+        ForwardableEmailMessage(val.unchecked_into())
+    }
+
+    fn unchecked_from_js_ref(val: &JsValue) -> &Self {
+        unsafe { &*(val as *const JsValue as *const Self) }
+    }
+}
+
+impl AsRef<JsValue> for ForwardableEmailMessage {
+    fn as_ref(&self) -> &JsValue {
+        &self.0
+    }
+}
+
+impl From<JsValue> for ForwardableEmailMessage {
+    fn from(val: JsValue) -> Self {
+        ForwardableEmailMessage(val.unchecked_into())
+    }
+}
+
+impl From<ForwardableEmailMessage> for JsValue {
+    fn from(sec: ForwardableEmailMessage) -> Self {
+        sec.0.into()
+    }
+}
+
+impl From<ForwardableEmailMessageExt> for ForwardableEmailMessage {
+    fn from(value: ForwardableEmailMessageExt) -> Self {
+        ForwardableEmailMessage(value)
+    }
+}
+
+impl ForwardableEmailMessage {
+    // method from is renamed compared to the JS version, because `from` conflicts with the From trait
+    // to was also renamed for consistency
+    pub fn from_envelope(&self) -> String {
+        self.0.from()
+    }
+
+    pub fn to_envelope(&self) -> String {
+        self.0.to()
+    }
+
+    pub fn raw(&self) -> Result<ByteStream, JsValue> {
+        Ok(ByteStream {
+            inner: wasm_streams::ReadableStream::from_raw(self.0.raw().dyn_into()?).into_stream(),
+        })
+    }
+
+    pub async fn raw_bytes(&self) -> Result<Vec<u8>, Error> {
+        self.raw()?
+            .try_fold(Vec::new(), |mut bytes, mut chunk| async move {
+                bytes.append(&mut chunk);
+                Ok(bytes)
+            })
+            .await
+    }
+
+    pub fn raw_size(&self) -> u32 {
+        self.0.raw_size()
+    }
+
+    pub fn set_reject(&self, reason: &str) -> Result<(), JsValue> {
+        self.0.set_reject(reason)
+    }
+
+    pub async fn forward(&self, rcpt_to: &str, headers: Option<Headers>) -> Result<(), JsValue> {
+        JsFuture::from(self.0.forward(rcpt_to, headers.unwrap_or_default().0)?).await?;
+        Ok(())
+    }
+
+    pub async fn reply(&self, message: EmailMessage) -> Result<(), JsValue> {
+        JsFuture::from(self.0.reply(message.0)?).await?;
```

Comment:
You may want to wrap these async calls as follows, otherwise the method's Future will not be Send:
```rust
let promise = self.0.reply(message.0)?;
let fut = SendFuture::new(JsFuture::from(promise));
fut.await?;
```

Basically, you don't want to hold `JsFuture` across `await` boundary.